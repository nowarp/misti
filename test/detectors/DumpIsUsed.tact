fun test1() {
    // Suspicious: Report debug prints
    let a: Int = 42;
    dump(a);

    // OK: Primitive literals are not reported
    dump(42);
    dump("test is called");
    dump(false);
    dump(null);
}
