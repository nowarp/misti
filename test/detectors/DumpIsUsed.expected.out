[INFO] DumpIsUsed: Found `dump` usage
test/detectors/DumpIsUsed.tact:4:5:
  3 |     let a: Int = 42;
> 4 |     dump(a);
          ^
  5 | 
Help: Using `dump` in production code can sometimes indicate complex code that requires additional review
See: https://nowarp.io/tools/misti/docs/detectors/DumpIsUsed